fundingmanager.go

```diff
@@ -277,7 +302,72 @@ func (f *fundingManager) Start() error {
 		f.barrierMtx.Unlock()
 
 		doneChan := make(chan struct{})
-		go f.waitForFundingConfirmation(channel, doneChan)
+		timeoutChan := make(chan struct{})
+
+		go func(ch *channeldb.OpenChannel) {
+			go f.waitForFundingWithTimeout(ch, doneChan, timeoutChan)
+
+			select {
+			case <-timeoutChan:
+				// forget the channel, delete it from the database
+				closeInfo := &channeldb.ChannelCloseSummary{
+					ChanPoint: *ch.FundingOutpoint,
+					RemotePub: ch.IdentityPub,
+					CloseType: channeldb.FundingCanceled,
+				}
+				ch.CloseChannel(closeInfo)
+			case <-f.quit:
+				// shutting down, will resume wait on startup
+			case <-doneChan:
+				// success, funding transaction confirmed
+			}
+		}(channel)
+	}
+
+	// Fetch all our open channels, and make sure they all finalized the opening
+	// process
+	openChannels, err := f.cfg.Wallet.ChannelDB.FetchAllChannels()
+	if err != nil {
+		return err
+	}
+
+	for _, channel := range openChannels {
+		channelState, shortChanID, err := f.getChannelOpeningState(channel.ChanID)
+		if err == channeldb.ErrChannelNotFound {
```

Comment:
I thought errors should all be gathered in `errors.go`, but since this one is in package `channeldb` this makes in hindsight not very much sense. Moved the error to `fundingManager.go`.